tests/bluetooth/host/keys/bt_keys_add_type/src/main.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/zephyr.h>
+#include <bluetooth/bluetooth.h>
+#include "testing_common_defs.h"
+#include <keys.h>
+#include "host_mocks/assert.h"
+#include "mocks/rpa.h"
+#include "mocks/keys_help_utils.h"
+
+/* This LUT contains different combinations of ID, Address and key type.
+ * Item in this list will be used to fill keys pool.
+ */
+static const struct id_addr_type testing_id_addr_type_lut[CONFIG_BT_MAX_PAIRED] = {
```

Comment:
User-configurable parameter (max pairing records in key pool) used with an array with an initializer list. This has a good chance of causing spontaneous compile errors with any implementation-specific selection of max pairing records.

Reply:
Regarding defining the array size within the initializer, we agreed before that this isn't a good practice.

But, this parameter `CONFIG_BT_MAX_PAIRED` is configured per UT project to satisfy a specific target and I won't consider it a `User-configurable parameter` as it isn't a generic project.
If someone is going to edit this parameter for a specific UT, he should ensure that the UT is still working.